force-app/main/default/lwc/joobleJobResults/joobleJobResults.js

```diff
@@ -0,0 +1,128 @@
+import { LightningElement, wire } from 'lwc';
+import { MessageContext, subscribe, unsubscribe } from 'lightning/messageService';
+import JOOBLE_LMS from '@salesforce/messageChannel/joobleLMS__c';
+import saveToSalesforce from '@salesforce/apex/JoobleSaveToSalesforce.saveToSalesforce';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+
+const columns = [
+    { label: 'Title', fieldName: 'title' },
+    { label: 'Location', fieldName: 'location' },
+    { label: 'Description', fieldName: 'snippet', wrapText: true },
+    { label: 'Company', fieldName: 'company' },
+];
+
+export default class JoobleJobResults extends LightningElement {
+
+    @wire (MessageContext) messageContext;
+    subscription = null;
+    noSubscription = true;
+    data = [];
+    columns = columns;
+    noResults = false;
+    savedRecords = [];
+    payload = '';
+
+    connectedCallback(){
+        this.handleSubscribe();
+
+        console.log('*** SUBSCRIPTION IN CALLBACK:', this.subscription);
+        console.log('*** NO SUBSCRIPTION IN CALLBACK:', this.noSubscription);
+    }
+
+    disconnectedCallback(){
+        this.handleUnsubscribe();
+        
+    }
+
+    handleSubscribe() {
+        if (!this.subscription) {
+            this.subscription = subscribe(
+                this.messageContext, 
+                JOOBLE_LMS, 
+                (message) => this.handleMessage(message)
+            );
+
+            
+            console.log('*** SUBSCRIPTION IN HANDLE:', this.subscription);
+            console.log('*** NO SUBSCRIPTION IN HANDLE:', this.noSubscription);
+        }
+    }
+
+    handleMessage(message) {
+        if (message) {
+            const responseData = JSON.parse(message.payload);
+            this.data = responseData.jobs.map(job => ({
+                company: job.company,
+                id: job.id,
+                link: job.link,
+                location: job.location,
+                salary: job.salary,
+                snippet: this.getPlainTextFromHTML(job.snippet),
+                source: job.source,
+                title: job.title,
+                type: job.type,
+                updated: job.updated
+            }));
+
+            console.log('*** RESPONSE DATA:', responseData);
+            console.log('*** DATA:', this.data);
+
+            this.noSubscription = false;
+            this.noResults = responseData.totalCount === 0;
+        }
+    }
+
+    handleUnsubscribe(){
+        if (this.subscription) {
+            unsubscribe(this.subscription);
+            this.subscription = null;
```

Comment:
Ensure that the subscription state (this.subscription and this.noSubscription) is consistently updated and used throughout the component's lifecycle to accurately reflect the subscription status.